### PR TITLE
feat: add styleExtensions

### PR DESCRIPTION
**Pull request by TheAlexLichter**
In your `css` array, you have to write the corresponding extension when providing your stylesheets. While transitioning from SCSS to PostCSS in several projects, I thought: "Why not completely omitting the extension as we do it for JS resolves :thinking:?"

There we go! 

## Types of changes
- [x] New feature (a non-breaking change which adds functionality)

## Checklist:
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [x] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

File: packages/core/src/resolver.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ export default class Resolver {
     return resolve(this.options.srcDir, path)
   }
 
-  resolvePath(path, { alias, module } = {}) {
+  resolvePath(path, { alias, module, isStyle } = {}) {
```
Comment (manniL) [inline packages/core/src/resolver.js]: Maybe we should move to `isXYZ` syntax generally for booleans here? :thinking:

**Review by galvez — DISMISSED** [review]
LGTM!

File: packages/core/src/resolver.js (hunk 1 above)
Comment (pi0) [reply packages/core/src/resolver.js]: I like this convention :+1:

File: packages/config/src/options.js [hunk 2]
```diff
@@ -43,6 +43,21 @@ export function getNuxtConfig(_options) {
...
+  if (!options.styleExtensions) {
+    options.styleExtensions = styleExtensionDefaults
+  }
+
+  if (typeof options.styleExtensions === 'function') {
+    // Execute functions with defaults passed in as first argument
+    options.styleExtensions = options.styleExtensions(styleExtensionDefaults)
+  }
+
+  if (typeof options.styleExtensions === 'string') {
```
Comment (pi0) [inline packages/config/src/options.js]: Why supporting such rare types and adding complexity?

File: packages/config/src/options.js [hunk 3]
```diff
@@ -43,6 +43,21 @@ export function getNuxtConfig(_options) {
...
 
+  const styleExtensionDefaults = ['css', 'pcss', 'postcss', 'styl', 'stylus', 'scss', 'sass', 'less']
+
+  if (!options.styleExtensions) {
+    options.styleExtensions = styleExtensionDefaults
+  }
+
+  if (typeof options.styleExtensions === 'function') {
+    // Execute functions with defaults passed in as first argument
+    options.styleExtensions = options.styleExtensions(styleExtensionDefaults)
```
Comment (pi0) [inline packages/config/src/options.js]: If the function returns nothing (undefined) we will get into trouble

File: packages/config/src/options.js [hunk 4]
```diff
@@ -43,6 +43,21 @@ export function getNuxtConfig(_options) {
     options.extensions = [options.extensions]
   }
 
+  const styleExtensionDefaults = ['css', 'pcss', 'postcss', 'styl', 'stylus', 'scss', 'sass', 'less']
```
Comment (pi0) [inline packages/config/src/options.js]: If defaults ordering would be important I suggest mentioning it in the docs too :)

**Review by pi0 — COMMENTED** [review]
Nice enhancement. But I don't see any reason adding complexity for handling option as a function or string unless a specific use case exists. replace or append merge strategy would be enough and more consistent. Modules has the opprotounity to modify options in any manner they need :)

File: packages/config/src/options.js (hunk 2 above)
Comment (manniL) [reply packages/config/src/options.js]: TBH: I mimicked that behavior from the `extensions` property. :see_no_evil: 

If it's not needed, I can remove it for sure :)

File: packages/config/src/options.js (hunk 3 above)
Comment (manniL) [reply packages/config/src/options.js]: Agreed. Maybe even remove the function check (and possible function usage) completely? :thinking:

File: packages/config/src/options.js (hunk 4 above)
Comment (manniL) [reply packages/config/src/options.js]: Yeah, I'll create a doc PR afterwards anyway :relaxed: 
Thanks for the heads up!

**Review by pi0 — DISMISSED** [review]

**Review by pi0 — APPROVED** [review]

**Pull request merged**